### PR TITLE
Implemented the ThumbNav component

### LINKED ISSUE #537: [FEATURE] Thumbnav component
### Describe the solution you'd like
Create a Thumbnav component based on the original Bootstrap Italia Thumbnav component: https://italia.github.io/bootstrap-italia/docs/menu-di-navigazione/thumbnav/

* [ ] Create the basic component
* [ ] Cover all the variants
* [ ] Create a Storybook set of stories based on the BI documentation

Notes:

* use BI classes to style the component.
* no additional CSS is expected in this component, unless justified shortcomings of the original CSS.

### PR DESCRIPTION
**Pull request by ffranchina**
Fixes #537.

#### PR Checklist

- [x] My branch is up-to-date with the Upstream `master` branch.
- [x] The unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

#### Short description of what this resolves:
I've implemented the component for the ThumbNav and added two examples to show the usage of the component.

**NOTE**: Currently the ThumbNav CSS code is not working on Firefox (see the issue https://github.com/italia/bootstrap-italia/issues/465 and my PR that fixes it https://github.com/italia/bootstrap-italia/pull/493) therefore make sure to test the react component on Chrome-based browsers.

### CODE REVIEW

File: src/components/ThumbNav/ThumbNav.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+      'thumb-nav-top': this.props.position === 'top',
+      'thumb-nav-right': this.props.position === 'right',
+      'thumb-nav-bottom': this.props.position === 'bottom',
+      'thumb-nav-left': this.props.position === 'left',
+
+      'thumb-nav-auto': this.props.rowItems !== undefined,
+      'thumb-nav-auto-2': this.props.rowItems === 2,
+      'thumb-nav-auto-3': this.props.rowItems === 3,
+      'thumb-nav-auto-4': this.props.rowItems === 4,
+      'thumb-nav-auto-5': this.props.rowItems === 5
```
Comment (dej611) [inline src/components/ThumbNav/ThumbNav.js:43]: Maybe we can compact this long lists:
```js
const classes = classNames('thumb-nav', {
      'thumb-nav-fixed': this.props.fixedWidth,
      'thumb-nav-small': this.props.isSmall,
      'thumb-nav-vertical': this.props.isVertical,
      'thumb-nav-nozoom': this.props.noZoomOnHover,
      [`thumb-nav-${this.props.overlayOnHover}`]: this.props.overlayOnHover,
      [`thumb-nav-${this.props.position}`]: this.props.position,
      [`thumb-nav-auto`]: this.props.rowItems != null,
      [`thumb-nav-auto-${this.props.rowItems}`]: typeof this.props.rowItems === 'number'
 });
```

**Comment by dej611** [comment]
Good job! Thanks @ffranchina ! 🎉 
Just one minor change: could you translate in italian the markdown content?

**Comment by ffranchina** [comment]
Thanks @dej611!
I'm applying your suggestions!
Shall I commit the directories `__snapshots__` as well?

File: src/components/ThumbNav/ThumbNav.js (hunk 1 above)
Comment (ffranchina) [reply src/components/ThumbNav/ThumbNav.js]: Thanks :)

**Comment by dej611** [comment]
> Thanks @dej611!
> I'm applying your suggestions!
> Shall I commit the directories `__snapshots__` as well?

Yes. Then when CircleCI will be back it could run tests again.

**Comment by ffranchina** [comment]
I'm not familiar with this `storyshot` and I'm not sure I did it correctly: it changed every single `__snapshots__` dir of the project.
I've committed it so you can have a look, sorry for annoying you with this small issue.

**Comment by dej611** [comment]
All good. Snapshot testing is quite noisy in general. 👍 
Great job!

**Review by dej611 — APPROVED** [review]

**Comment by ffranchina** [comment]
Great! Good to know! :+1:

**Pull request merged**